File: python/rotational-cipher/rotational_cipher.py
def rotate(text, key):
    def encode_single(char, key):
        if char.isupper():
            return chr((ord(char) - 65 + key) % 26 + 65)
        elif char.islower():
            return chr((ord(char) - 97 + key) % 26 + 97)
        else:
            return char

    return ''.join(map(lambda c: encode_single(c, key), text))

File: haskell/rotational-cipher/src/RotationalCipher.hs
module RotationalCipher
  ( rotate
  ) where

import           Data.Char                      ( chr
                                                , isLower
                                                , isUpper
                                                , ord
                                                )

rotate :: Int -> String -> String
rotate k = map encodeChar
 where
  encodeChar c | isUpper c = chr ((ord c - 65 + k) `mod` 26 + 65)
               | isLower c = chr ((ord c - 97 + k) `mod` 26 + 97)
               | otherwise = c
